system/HTTP/CURLRequest.php

```diff
@@ -387,6 +387,10 @@ public function send(string $method, string $url)
             $output = substr($output, strpos($output, $breakString) + 4);
         }
 
+        if (strpos($output, 'HTTP/1.1 200 Connection established') === 0) {
```

Comment:
It may be "Connection Established".

Is it always "HTTP/1.1" ?

Reply:
@kenjis You are right, I'm studying now whole concept behind this and it seems that there should be different headers. 

The original tunnelling spec required the proxy to return a `200 Connection Established` to the client once it had successfully connected to the server, which is what we're seeing. This is potentially followed by more headers, and then an empty line, before the connection becomes transparent and we get the actual response from the server. So, we get two sets of headers. RFC 2817 relaxes this, and allows any 2xx response to a CONNECT request.

Some very late clarification. When you connect to an SSL/TLS server through a proxy, the proxy establishes a tunnel, using HTTP CONNECT. This is covered in [RFC2817](http://tools.ietf.org/rfc/rfc2817) and the expired [tunneling spec](http://www.web-cache.com/Writings/Internet-Drafts/draft-luotonen-web-proxy-tunneling-01.txt), and not RFC261.

When making an https request through a proxy, a CONNECT request is of 
course made to the proxy before the SSL connection can be made with the 
remote server. 

Currently, the proxy's response to the CONNECT is written to the libcurl 
callback function like the rest of the response, such that it appears 
before the remote server's response headers. So, for instance, to my 
client code, the response looks like this: 

```sh
HTTP/1.1 200 Connection established <-- proxy's response 
Proxy-Agent: Fortinet-Proxy/1.0
```

```sh
HTTP/2 200 <-- remote server's response 
date: Mon, 26 Jun 2023 07:44:50 GMT 
etc 
```

some proxies are rather minimalist and only give: 

```sh
HTTP/1.0 200 <-- proxy's response 
```

```sh
HTTP/1.0 200 OK <-- remote server's response 
Date: Tue, 04 Oct 2005 20:17:22 GMT 
etc 
```

The best method I have thought of for detecting whether the response 
contains a header like this (with a proxy's CONNECT response first) is 
to set a header callback and see if I get an empty header line in the 
middle of the headers. I haven't actually tested this solution yet.